pxr/usd/lib/sdf/fileFormat.cpp

```diff
@@ -257,13 +258,39 @@ SdfFileFormat::GetFileExtension(
     return extension.empty() ? s : extension;
 }
 
+/* static */
+std::set<std::string>
+SdfFileFormat::FindAllFileFormatExtensions()
+{
```

Comment:
I think this code will crash if the plugins for the file formats returned by FindAllFileFormats are not loaded first. In that case, FindAllFileFormats will return TfTypes that have been declared in plugInfo.json files but not yet defined by a call to TfType::Define, which means they won't have a factory set, which means factory->New() will die.

Reply:
Also, ideally we would not need to actually load the file format plugins to determine the available extensions, since that's all declared in plugInfo.json files. In other words, if I were to import Sdf in Python and call this function, it shouldn't cause Usd to load.